.github/workflows/build-main.yml

```diff
@@ -0,0 +1,66 @@
+name: Build Main/Master SNAPSHOT
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "5 6 * * *"
+  push:
+    branches:
+      - 'main'
+      - 'master'
+jobs:
+  build-master:
+    name: Build & Package Master
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Java for publishing to GitHub Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: github
+
+      # Version artifact based off of branch and commit SHA.
+      - name: Version Artifact
+        run: mvn versions:set -DnewVersion="master-SNAPSHOT"
+
+      # Publish to GitHub Packages
+      - name: Publish package
+        run: mvn -B clean deploy -DskipTests=true
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  build-sha:
+    name: Build & Package SHA
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Java for publishing to GitHub Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          cache: 'maven'
+          server-id: github
+
+      # Version artifact based off of branch and commit SHA.
+      - name: Version Artifact
+        run: mvn versions:set -DnewVersion="$(git rev-parse --short HEAD)-SNAPSHOT"
```

Comment:
You might be able to use `${{github.sha}}` here.

Reply:
@StevenMassaro doesn't that give the full commit has and not the short version?

Reply:
Good point, it does give the full hash.